CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+cmake_minimum_required(VERSION 3.8)
+project(task-dispatcher)
+
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
this should not be in libraries, only in the final app.
we should only test for it being set, not set it ourself